src/material-examples/tree-checklist/tree-checklist-example.ts

```diff
@@ -168,10 +168,21 @@ export class TreeChecklistExample {
     return flatNode;
   }
 
-  /** Whether all the descendants of the node are selected */
+  /**
+   * Whether all the descendants of the node are selected.
+   * And remove the node from this.checklistSelection if it isn't selected.
+   */
   descendantsAllSelected(node: TodoItemFlatNode): boolean {
     const descendants = this.treeControl.getDescendants(node);
-    return descendants.every(child => this.checklistSelection.isSelected(child));
+    const descAllSelected = descendants.every(child =>
+      this.checklistSelection.isSelected(child)
```

Comment:
It looks like the `this.checklistSelection` is being repeated in this function a lot and it's somewhat long. Consider moving it into a constant like `const selection = this.checklistSelection`.